util/testkit/testkit.go

```diff
@@ -147,12 +148,27 @@ func (tk *TestKit) Exec(sql string, args ...interface{}) (sqlexec.RecordSet, err
 	}
 	ctx := context.Background()
 	if len(args) == 0 {
-		var rss []sqlexec.RecordSet
-		rss, err = tk.Se.Execute(ctx, sql)
-		if err == nil && len(rss) > 0 {
-			return rss[0], nil
+		stmts, warns, err := tk.Se.Parse(ctx, sql)
+		if err != nil {
+			return nil, errors.Trace(err)
 		}
-		return nil, errors.Trace(err)
+		var rs0 sqlexec.RecordSet
+		for i, stmt := range stmts {
+			rs, err := tk.Se.ExecuteStmt(ctx, stmt)
+			if i == 0 {
+				rs0 = rs
+			}
+			if err != nil {
+				tk.Se.GetSessionVars().StmtCtx.AppendError(err)
+				return nil, errors.Trace(err)
+			}
+		}
+		if len(warns) > 0 {
+			for _, warn := range warns {
```

Comment:
Why append warnings after execute?

Reply:
A test fails here, something like `select /*+wrong index merge hint?*/ xxx`
This one should execute success and report warning that the hint is ignored.

We can't add the warning immediately after `Parse`,  
because in  the following `ExecuteStmt`, 
the statement context  is  reset and warning info would lost,
so the warn is append after the statement finish.